### DISCUSSION TITLE
Windows 10 boot from UEFI

### DISCUSSION THREAD

**Discussion by PatcoreShare** [ipxe/ipxe#305] [Q&A]
Hi, I can't boot from UEFI  Windows 10 1803 
Have someone  solusion? 

Boot from legacy is working
Boot from UEFI isn't working 

![1](https://user-images.githubusercontent.com/11398415/112971790-dc560180-914f-11eb-96e2-508fbb1f7d1b.png)
![2](https://user-images.githubusercontent.com/11398415/112971795-dcee9800-914f-11eb-8c70-ad04500175ce.png)
![3](https://user-images.githubusercontent.com/11398415/112971800-de1fc500-914f-11eb-98e2-74543694343c.png)

My config

**dhcpd.conf**
`````

# dhcpd.conf
option domain-name "example.org";
option domain-name-servers ns1.example.org, ns2.example.org;

default-lease-time 600;
max-lease-time 7200;

ddns-update-style none;

authoritative;
allow booting;
allow bootp;

subnet 10.10.10.0 netmask 255.255.255.0 {
  interface eth1;
  range 10.10.10.26 10.10.10.100;
  option domain-name-servers ns1.internal.example.org;
  option domain-name "internal.example.org";
  option subnet-mask 255.255.255.0;
  option routers 10.10.10.1;
  option broadcast-address 10.10.10.255;
  default-lease-time 600;
  max-lease-time 7200;
}

host win-mother{
 next-server 10.10.10.10;
#Wyse
# hardware ethernet F0:D4:E2:EF:5C:56;
#VM
 hardware ethernet F0:D4:E2:EF:5C:55;
 if exists user-class and option user-class = "iPXE"{
  filename "http://10.10.10.10/winpe/boot.ipxe";

 } else{
#  filename "ipxe.efi";
 filename "ipxe.efi";
#  filename "ipxe.kpxe";
#  filename "undionly.kpxe";
 } 
}
`````

 

**boot.ipxe**
`````
#!ipxe
cupid --ext 29 && set arch amd64 

set base http://10.10.10.10/winpe

sanhook iscsi:10.10.10.10:::1:iqn.windows:mother

kernel wimboot
initrd -n bootmgr.efi	${arch}/media/bootmgr.efi	bootmgr.efi 
initrd -n bcd	${arch}/media/Boot/BCD	bcd 
initrd -n boot.sdi	${arch}/media/Boot/boot.sdi	boot.sdi 
initrd -n boot.wim  ${arch}/media/sources/boot.wim	boot.wim
boot
`````

**Comment by NiKiZe** [comment]
From the images the boot seem to be working just fine?
The issue with windows setup not being able to find installation media is more of a Windows setup issue than iPXE.

In the first stage before clicking start setup, have you tried shift+F10 and verified with diskpart list volume that the disk is there? If not have you tried to start the network?

**Reply by PatcoreShare** [reply]
Diskpart 
![Screenshot_1](https://user-images.githubusercontent.com/11398415/112981000-e16c7e00-915a-11eb-92d8-3df79ef55727.png)

ipconfig
![Screenshot_2](https://user-images.githubusercontent.com/11398415/112981021-e7faf580-915a-11eb-8ee2-9055d6356600.png)

**Reply by NiKiZe** [reply]
So you have one disk, is it installation source or installation target?

**Reply by PatcoreShare** [reply]
I share folder on samba
(https://ipxe.org/howto/winpe)
![Screenshot_1](https://user-images.githubusercontent.com/11398415/113009264-13d8a400-9178-11eb-989b-f46fdcae0258.png)

**Reply by NiKiZe** [reply]
And you started the setup from that share as I that picture?

**Reply by PatcoreShare** [reply]
Correct

**Reply by NiKiZe** [reply]
Is this at the begining of setup, or is it after disk ha been configured, and files has been copied?

**Reply by PatcoreShare** [reply]
Here is a short video of trouble:
https://youtu.be/q2g-IjQ5-C4

**Comment by JBWinnipeg** [comment]
That appears to be an issue with the drivers loading (probably the default ones that come with the stock Win8 image) into the WinPE environment.  Your boot.wim (or boot.iso) doesn't have the required network drivers that supports whatever network card your VM is using so when you try to connect to your network share of your OS deployment image, it fails and you get that error.

You'll need to look into how to inject the requisite network drivers into the pre-boot environment (that launches through PXE) so that it can see the OS install files that it's looking for when the setup process launches.

**Reply by NiKiZe** [reply]
Good catch, maybe the PE image is not same version (newer win 10) than the OS being installed (win 8)

**Reply by JBWinnipeg** [reply]
All this to say, the issue doesn't appear to be related to iPXE at all since you're clearly past the point where PXE is controlling things.  Unless I'm misunderstanding things, in which case definitely more info required!

**Reply by NiKiZe** [reply]
Agreed, not an iPXE issue, but as it is in discussion I wanted to try and help @patriocoz out, and maybe get it documented for anyone else coming around.

**Reply by PatcoreShare** [reply]
I don't know what drivers should I upload. I use Hyper-v. 
When I use: 
![image](https://user-images.githubusercontent.com/11398415/113664981-ff7f3480-96ac-11eb-9601-1064c472712f.png)
successful installation
When I use:
![image](https://user-images.githubusercontent.com/11398415/113665028-132a9b00-96ad-11eb-8d6c-a25757d513b5.png)
error as above


use drivers from the physical machine?

**Comment by PatcoreShare** [comment]
I migrate to vmware
I have problem with disk initialization 
![image](https://user-images.githubusercontent.com/11398415/114390407-842eee80-9b96-11eb-80e3-c688f173b3d6.png)

Page don't exist 
https://ipxe.org/er/3c22208
https://ipxe.org/3c22208

The same configuration

**Reply by sebaxakerhtc** [reply]
Does your VM have another storage? 
Because 0x80 is local drive.
Try this:
`sanhook --drive 0x81 iscsi....`

**Reply by mcb30** [reply]
@patriocoz The error message wraps onto the next line: the relevant URL is http://ipxe.org/3c222083

**Reply by PatcoreShare** [reply]
![image](https://user-images.githubusercontent.com/11398415/114393272-20a6c000-9b9a-11eb-8be3-eb8a7ee322bb.png)
![image](https://user-images.githubusercontent.com/11398415/114393308-28fefb00-9b9a-11eb-9aef-dc71fcf8d001.png)
Computer without disk
ipxe.efi download from: http://boot.ipxe.org/ipxe.efi

**Reply by sebaxakerhtc** [reply]
`You can work around the bug in your UEFI firmware using the --no-describe option of the sanboot command` from link provided by @mcb30 
replase `--drive 0x81` with `--no-describe`
But can you show us more? i.e. what is the san device, and what are the filesystem, partition table (GPT or MBR), and files on it?
How was installed this system or it is the install media?

**Reply by mcb30** [reply]
I've updated the text at http://ipxe.org/3c222083 to point out that switching from UEFI to "legacy" BIOS mode will fix the problem (since the bug affects only UEFI mode operation).

**Reply by sebaxakerhtc** [reply]
> I've updated the text at http://ipxe.org/3c222083 to point out that switching from UEFI to "legacy" BIOS mode will fix the problem (since the bug affects only UEFI mode operation).

Does that means we can't sanhook iscsi:hdd to VMWare VM? I need to try it :)

**Reply by PatcoreShare** [reply]
It is the install media. 
I testing in VMWare.
When i use` --no-describe` I can;t see drive for install Windows.
ie: share folder 
![image](https://user-images.githubusercontent.com/11398415/114394818-edfdc700-9b9b-11eb-8889-345652e2e018.png)

![image](https://user-images.githubusercontent.com/11398415/114395101-4c2aaa00-9b9c-11eb-903e-b53d3cdcc353.png)

**Reply by sebaxakerhtc** [reply]
For local installation I use the WinPE + Installer from smb. For me it is the best solution.
But for world's(internet) installation i use WinPE + iscsi.
This works for me on any PC and VM

**Reply by PatcoreShare** [reply]
I share folder on Windows Server, only for test

> I've updated the text at http://ipxe.org/3c222083 to point out that switching from UEFI to "legacy" BIOS mode will fix the problem (since the bug affects only UEFI mode operation).

I have wyse 5070, Wyse 5070 don't support Legacy. How I can use iPXE? 
Any Idea? 

>If you are unable to switch to “legacy” BIOS mode, then you will need to contact your UEFI firmware vendor for a fix.

OK :)

**Reply by sebaxakerhtc** [reply]
> How I can use iPXE?
> Any Idea?

Don't you like my solution to boot WinPE over http and install Windows from iSCSI? Or you want to boot from iscsi only?


> Wyse 5070 don't support Legacy

Your problem have a place just on VMWare. May be on Wyse 5070 you can sanhook without problems. Try it.

**Reply by sebaxakerhtc** [reply]
Latest Windows 10 images have a size of install.wim more than 3999 Mb. That means you can't format install media in Fat32 and place there install.wim because of Fat32 limitations. You should split install.wim to istall.swm files, or to use install media formated in other Filesystem (i.e. NTFS/ExtFS). But this way you can't boot from it in UEFI mode...

**Reply by PatcoreShare** [reply]
>Don't you like my solution to boot WinPE over http and install Windows from iSCSI? Or you want to boot from iscsi only?

Boot from iscsi only, diskless client. 


>Your problem have a place just on VMWare. May be on Wyse 5070 you can sanhook without problems. Try it.


I'm testing on a virtual machine to learn the capabilities. 

I will try to test on wyse 5070
Thank you all for your help

**Comment by JBWinnipeg** [comment]
You should go back to your original method.  You were getting closer!  The issue appears to be that your WinPE environment, doesn't have the Hyper-V network drivers built into it.  You need to bundle the Hyper-V drivers into the WinPE image for the boot.iso file that PXE is loading.

Here's a link for a way to extract the drivers you're looking for, if you ever go back to your Hyper-V solution.

https://weblogs.asp.net/owscott/creating-winpe-with-integration-services-for-hyper-v

Once your WinPE / Windows Setup environment has the Hyper-V drivers, I'd wager you'll be able to see and use the Samba share you've setup (provided you've got the pathing correct in the Deployment task list)
